src/ret_val.h

```diff
@@ -16,28 +19,37 @@
  * as defaults for the success/failure values respectively.
  */
 
+template<typename S>
+using _is_convertible_to_string =
```

Comment:
[cppreference.com](http://en.cppreference.com/w/cpp/language/identifiers) says identifiers starting with an underscore are reserved *in the global namespace*. (Put it into namespace `cata`?)

Reply:
Or just remove the underscore.